core/templates/dev/head/domain/exploration/EditableExplorationBackendApiService.js

```diff
@@ -0,0 +1,163 @@
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: String(explorationId)
+        });
+
+      params = {};
+      if (applyDraft) {
+        params.apply_draft = applyDraft;
+      }
+      if (v || v === 0) {
+        params.v = v;
+      }
+
+      $http.get(explorationDataUrl, {
+        params: params
+      }).then(function(response) {
+        var exploration = angular.copy(response.data);
+        if (successCallback) {
+          successCallback(exploration);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _updateExploration = function(
+        explorationId, explorationVersion, commitMessage, changeList,
+        successCallback, errorCallback) {
+      var editableExplorationDataUrl = (
+        UrlInterpolationService.interpolateUrl(
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId
+        })
+      );
+
+      var putData = {
+        version: explorationVersion,
+        commit_message: commitMessage,
+        change_list: changeList
+      };
+      $http.put(editableExplorationDataUrl, putData).then(function(response) {
+        // The returned data is an updated exploration dict.
+        var exploration = angular.copy(response.data);
+
+        // Update the ReadOnlyExplorationBackendApiService's cache with the
+        // new exploration.
+        ReadOnlyExplorationBackendApiService.cacheExploration(
+          explorationId, exploration);
+
+        if (successCallback) {
+          successCallback(exploration);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _deleteExploration = function(
+        explorationId, role, successCallback, errorCallback) {
+      var editableExplorationDataUrl = (
+        UrlInterpolationService.interpolateUrl(
+        EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId
+        })
+      );
+
+      params = {};
+      if (role) {
+        params.role = JSON.stringify(role);
+      }
+
+      $http['delete'](editableExplorationDataUrl, {
+        params: params
+      }).then(function(response) {
+        // What does this return? What happens on Fail?
+        // What happens wtih ReadOnlyExplorationBackendApiService
+        // on success or fail.
+
+        // Delete item from the ReadOnlyExplorationBackendApiService's cache
+        ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
+          explorationId);
+
+        if (successCallback) {
+          successCallback(response.data);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    return {
+      fetchExploration: function(explorationId, applyDraft, v) {
```

Comment:
Let's keep this fetch for now, but make it explicit via documentation comments when to use one or the other.